src/backend/storage/course_mapper.rs

```diff
@@ -0,0 +1,38 @@
+use sqlx::{Error, FromRow, Row};
+use sqlx::postgres::PgRow;
+use sqlx::types::JsonValue;
+
+use crate::models::{AcademicLevel, Course, CourseOffering, Prerequisites};
+
+impl<'r> FromRow<'r, PgRow> for Course {
+    fn from_row(row: &'r PgRow) -> Result<Self, Error> {
+        Ok(Course {
```

Comment:
The course struct cannot derive the default implementation of FromRow since it need to call custom deserializer trait on course offering (projected json list of offerings per course) and for time stamps in each course schedule.